aptos-move/framework/aptos-framework/sources/dispatchable_fungible_asset.move

```diff
@@ -0,0 +1,129 @@
+        let store_obj = &object::generate_signer(constructor_ref);
+        move_to<TransferRefStore>(store_obj, TransferRefStore {
+            transfer_ref: fungible_asset::generate_transfer_ref(constructor_ref),
+        });
+    }
+
+    /// Withdraw `amount` of the fungible asset from `store` by the owner.
+    ///
+    /// The semantics of deposit will be governed by the function specified in DispatchFunctionStore.
+    public fun withdraw<T: key>(
+        owner: &signer,
+        store: Object<T>,
+        amount: u64,
+    ): FungibleAsset acquires TransferRefStore {
+        if(fungible_asset::is_dispatchable(store)) {
+            let func = fungible_asset::withdraw_function(store);
+            dispatchable_withdraw(
+                signer::address_of(owner),
+                store,
+                amount,
+                borrow_transfer_ref(store),
+                &func,
+            )
+        } else {
+            fungible_asset::withdraw(owner, store, amount)
+        }
+    }
+
+    /// Deposit `amount` of the fungible asset to `store`.
+    ///
+    /// The semantics of deposit will be governed by the function specified in DispatchFunctionStore.
+    public fun deposit<T: key>(
+        store: Object<T>,
+        fa: FungibleAsset
+    ) acquires TransferRefStore {
+        if(fungible_asset::is_dispatchable(store)) {
+            let func = fungible_asset::deposit_function(store);
+            dispatchable_deposit(
+                store,
+                fa,
+                borrow_transfer_ref(store),
+                &func
+            )
+        } else {
+            fungible_asset::deposit(store, fa)
+        }
+    }
+
+    inline fun borrow_transfer_ref<T: key>(
+        metadata: Object<T>
+    ): &TransferRef acquires TransferRefStore {
+        let metadata_addr = object::object_address(&fungible_asset::store_metadata(metadata));
+        assert!(exists<TransferRefStore>(metadata_addr), error::not_found(ESTORE_NOT_FOUND));
+        &borrow_global<TransferRefStore>(metadata_addr).transfer_ref
+    }
+
+    /// A transfer with a fixed amount debited from the sender
+    public fun transfer_fixed_send<T: key>(
+        _sender: &signer,
+        from: Object<T>,
+        to: Object<T>,
+        send_amount: u64,
+    ) acquires TransferRefStore {
+        let fa = fungible_asset::withdraw_with_ref(borrow_transfer_ref(from), from, send_amount);
+        deposit(to, fa);
+    }
+
+    /// A transfer with a fixed amount credited to the recipient
+    public fun transfer_fixed_receive<T: key>(
+        sender: &signer,
+        from: Object<T>,
+        to: Object<T>,
+        receive_amount: u64,
+    ) acquires TransferRefStore {
+        let fa = withdraw(sender, from, receive_amount);
+        fungible_asset::deposit_with_ref(borrow_transfer_ref(from), to, fa);
+    }
+
+    native fun dispatchable_withdraw<T: key>(
```

Comment:
new native functions need to be gated by a feature flag.

you can also gate fungible_asset::register_dispatch_functions with the same flag.

Reply:
Sounds good. Lemme add the feature gate.